### ISSUE TITLE
 Why custom attributes are added to the html tags instead of :root?

### ISSUE THREAD

**Issue by Merry909** [jhildenbiddle/css-vars-ponyfill#72]
Hi John Hildenbiddle,

I use the 'css-vars-ponyfill' in my angular project.
And now, we want to switch the theme by the router param.
And I find when I change the css3 variables , code as below

this.theme = 'default';
      if (params["theme"] != undefined && this.theme != params['theme']) {
        this.theme = params['theme'];
        cssVars({
          variables: {
            ...Themes[this.theme].properties
          }
        });
      }

The custom attributes are added on the html tag, not overridden those corresponding attributes of :root.

<html lang="en" style="--first-item-bgd:#010b10; --first-item-border:#ffc72a; --second-item-bgd:#19313b; --second-item-border:#33363d; --third-item-bgd:#1d2327; --third-item-border:#254d64;">

Do you have any way to show these properties on :root not html tag?
THX!

Merry

**Comment by jhildenbiddle** [comment]
@Merry909 --

Happy to help, but I don't understand your explanation of the issue. What custom attributes are you referring to?

Can you provide a demo for me to troubleshoot using either [stackblitz](https://stackblitz.com/) or [codesandbox](https://codesandbox.io)?

**Comment by Merry909** [comment]
Thanks for your reply.

such as:

this.theme = 'default';
if (params["theme"] != undefined && this.theme != params['theme']) {
this.theme = params['theme'];
cssVars({
variables: {
       "--first-item-bgd": "#fff",
        "--first-item-border": "#fff",
        "--second-item-bgd": "#fff",
        "--second-item-border": "#fff",
        "--third-item-bgd": "#fff",
        "--third-item-border": "#fff",
}
});
}

then the html tag will be rendered as:
<html style='--first-item-bgd: #fff; --first-item-border: #fff; ....'>

And I want change the :root  not html or not show inline style.

I am very sorry that due to the company management  and regulations, I have no access to upload my project . 
Please understand.
If needed, I will create a simple project for my question after work.

Thanks a lot!
Merry

**Comment by jhildenbiddle** [comment]
@Merry909 --

The ponyfill inserts `<style>` elements into the DOM. It does not add or modify `style` attributes values on existing elements (like the `<html>` tag).

The behavior you are seeing is a result of something else in your app.

**Comment by Merry909** [comment]
@jhildenbiddle 
This issue have been solved and I set the 'onlyLegacy' as 'false'.
Thanks!

**Comment by MiselKING** [comment]
Here is the [stackblitz](https://stackblitz.com/edit/angular-s8mmbx) example proving this. And yes, setting `onlyLegacy` to `false` solves this, but the question is whether this is expected behavior or not?

**Comment by jhildenbiddle** [comment]
@Merry909 @MiselKING --

Thank you for creating the stackblitz demo. It's makes resolving issues like this much easier.

I'll take a look at this early next week (as well as the other open issues). I stand by my earlier comment that the ponyfill itself is not adding the `style` tag with custom property values to the `<html>` element, but thanks to the demo I now have a better understanding of what you are trying to do and some potential changes that can be made to prevent Angular from doing what it's doing.

More to come...

**Comment by jhildenbiddle** [comment]
@Merry909 @MiselKING --

TL;DR: The ponyfill is not adding/modifying `<html>` element attributes, but the way the ponyfill is being called is causing Angular to do so.

Here's what's happening under the hood:

1. When the ponyfill is called with `onlyLegacy:true` (default) in a modern browser, the ponyfill iterates over the `options.variables` object and applies each value using [setProperty()](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/setProperty) on `document.documentElement.style`.
1. When Angular detects a change to the `document.documentElement` styles, Angular applies those styles to the `<html>` element via the `style` attribute.

Here is an easy way to see this in action:

1. Load you demo's preview in a separate window: https://angular-s8mmbx.stackblitz.io/
1. Open the developer tools
1. Select the `Elements` tab
1. If the console is not visible under the `Elements` tab, press `ESC`
1. Enter the following in the console: 

   ```js
   document.documentElement.style.cssText
   ```

   Notice that the output is the same as the `style` attribute of the `<html>` element:

   ```html
   <html class=" " style="--foreground:#FF0000;"><head>
   ```

1. Enter the following in the console: 

   ```js
   document.documentElement.style.setProperty('background', 'black')
   ```

   Notice that the value of the `style` attribute on the `<head>` element now includes a `background: black` declaration:

   ```html
   <html class=" " style="--foreground:#FF0000; background: black;"><head>
   ```

Obviously, changing the background color via the console has nothing to do with the ponyfill. Angular is observing `document.documentElement` style changes and, for whatever reason, applying those changes to the `<head>` element via the `style` attribute. 

The reason calling the ponyfill with `onlyLegacy:false` doesn't cause this issue is because in this scenario the ponyfill doesn't modify `document.documentElement` styles using [setProperty()](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/setProperty), opting instead to append new `<style>` elements with transformed CSS which is required for legacy browsers.

Interestingly, if you move the `import` and `cssVars()` call to `ponyfill.ts` you won't have this issue. I am assuming this is due to the fact that in this scenario the ponyfill call (and therefore the `document.documentElement` style changes) is happening at an application level rather than a component level. Again, I'm not an Angular expert, so this is just a guess.

Whew. A lengthy explanation, but there you go. Hope this helps.
